platform/qt/qmlapp/main.qml

```diff
@@ -61,51 +61,122 @@ ApplicationWindow {
                             property var url: "mapbox://styles/mapbox/streets-v9"
                         },
                         MapParameter {
+                            id: waterPaint
+                            property var type: "paint"
+                            property var layer: "water"
+                            property var fillColor: waterColorDialog.color
+                        },
+                        MapParameter {
                             property var type: "source"
-                            property var id: "testSource"
+                            property var name: "routeSource"
                             property var sourceType: "geojson"
                             property var data: ":source.geojson"
                         },
                         MapParameter {
                             property var type: "layer"
-                            property var id: "testLayer"
-                            property var layerType: "fill"
-                            property var source: "testSource"
+                            property var name: "routeCase"
+                            property var layerType: "line"
+                            property var source: "routeSource"
                         },
                         MapParameter {
                             property var type: "paint"
-                            property var layer: "testLayer"
-                            property var fillColor: Qt.rgba(0, 0, 1, 1)
+                            property var layer: "routeCase"
+                            property color lineColor: "white"
```

Comment:
Indeed - I must have confused this with some other error I had at the time I was implementing this - good to know Qt is smart enough on this ;)